image/dicom_le.ksy

```diff
@@ -0,0 +1,147 @@
+meta:
+  id: dicom
+  file-extension: dcm
+  endian: le
+seq:
+  - id: file_header
+    type: t_file_header
+  - id: dataset
+    type: t_dataset
+types:
+  t_file_header:
+    seq:
+      - id: preamble
+        size: 128
+      - id: magic
+        contents: 'DICM'
+  t_dataset:
+    seq:
+      - id: elements1
+        type: t_dataentry_state_implicit
+        if: not _io.eof
+        repeat: until
+        repeat-until: _io.eof or _.is_transfer_syntax_change_explicit
+      - id: elements2
+        type: t_dataentry_state_explicit
+        repeat: eos
+  t_dataentry_state_implicit:
+    seq:
+      - id: tag_group
+        type: u2
+      - id: tag_elem
+        type: u2
+      - id: header
+        type: t_entry_header_implicit
+        if: entry_implicit
+      - id: header
+        type: t_entry_header_explicit
+        if: not entry_implicit
```

Comment:
Yeah, I tried to use switching types, but then I could not read even the common fields of those types. I had a feeling that this is a hack, but I was going to use it from the webide anyway... Now that I know there is typecasting in the system I will look for ways to improve it so it works in other languages too